docs/examples/async_rpc.py

```diff
@@ -0,0 +1,6 @@
+with ClusterRpcProxy(config) as cluster_rpc:
+    hello_res = cluster_rpc.service_x.remote_method.async("hello")
+    world_res = cluster_rpc.service_x.remote_method.async("world")
+    # do work while waiting
+    hello_res.wait()  # "x-y-hello"
```

Comment:
i wonder if we could re-write these as doctests and run them (with say a pytest hook that spins up the example services)

this should be  `res.result()`, not `wait()`